py/bc.c

```diff
@@ -229,8 +228,17 @@ STATIC void mp_setup_code_state_helper(mp_code_state_t *code_state, size_t n_arg
                     MP_ERROR_TEXT("unexpected keyword argument '%q'"), MP_OBJ_QSTR_VALUE(wanted_arg_name));
                 #endif
             }
-            mp_obj_dict_store(dict, kwargs[2 * i], kwargs[2 * i + 1]);
-        continue2:;
+            mp_map_elem_t *elem = mp_map_lookup(mp_obj_dict_get_map(dict), wanted_arg_name, MP_MAP_LOOKUP_ADD_IF_NOT_FOUND);
+            if (elem->value == MP_OBJ_NULL) {
+                elem->value = kwargs[2 * i + 1];
+            } else {
+                goto error_multiple;
+            }
+        continue2:
+            continue;
+        error_multiple:
+            mp_raise_msg_varg(&mp_type_TypeError,
+                MP_ERROR_TEXT("function got multiple values for argument '%q'"), MP_OBJ_QSTR_VALUE(wanted_arg_name));
```

Comment:
Moving this error message here feels a little awkward... maybe keep it where it was and just add a `error_multiple:` label on line 216?

Reply:
done

Reply:
Very good!